### PR TITLE
I removed redirect authentication from the NavigationBar component, a…

### PR DESCRIPTION
**Pull request by salman90**
…lso changed the application title in PageLayout component

## Purpose

* Removing the redirect auth flow from the sample. Also changed the main title of the sample from Authentication Library For React - Tutorial to Azure AD Step-up Authentication Demo in the PageLayout component.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[] Feature
[x] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by v-michaelmi** [comment]
Just gave this a test drive.

LGTM

File: 6-AdvancedScenarios/3-call-api-acrs/API/views/dashboard.ejs [hunk 1]
```diff
@@ -38,15 +38,17 @@
...
                             <option value="DELETE">DELETE</option>
                             <option value="UPDATE">UPDATE</option>
                         </select>
                         <select class="form-select" name="authContext">
-                            <option selected>Select Auth Context</option>
-                            <% acrsList.forEach(ac=> { %>
-                                <option value="<%= ac.id %> <%= ac.displayName %>"><%= ac.displayName %></a>
+                            <% acrsList.forEach((ac, index)=> { %>
+                                <% if (index === 0) { %>
+                                    <option selected value="<%= ac.id %> <%= ac.displayName %>"><%= ac.displayName %></a>
```
Comment (v-michaelmi) [inline 6-AdvancedScenarios/3-call-api-acrs/API/views/dashboard.ejs:48]: Why are you checking for index 0 here?

Comment (v-michaelmi) [reply 6-AdvancedScenarios/3-call-api-acrs/API/views/dashboard.ejs:48]: I see this is to add the **selected** attirbute.

Purely cosmetic?

**Comment by kalyankrishna1** [comment]
>     "Sample": {

Please check why the full url to the app registration is not being printed for configure.ps1 

---
Refers to: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/sample.json:2 in 6101893. [](commit_id = 610189380c779fb416527f3be865befe8be554c4, deletion_comment = False)

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

**Review by aremo-ms — COMMENTED** [review]
Sample 6-3. 
The Configure.ps1 asks 
"optionalClaims": {
		"accessToken": [
			{
				"additionalProperties": [],
				"essential": false,
				"name": "xms_cc",
				"source": null
			}
		],
		"idToken": [],
		"saml2Token": []
	},

**Comment by aremo-ms** [comment]
> Sample 6-3. The Configure.ps1 asks to add xms_cc.
> 
> **The actual value there should be:**
> 
> ```json
> "optionalClaims": {
> 		"accessToken": [
> 			{
> 				"additionalProperties": [],
> 				"essential": false,
> 				"name": "xms_cc",
> 				"source": null
> 			}
> 		]
> 	}
> ```

Please disregard - this is actually inside README.md

**Review by aremo-ms — APPROVED** [review]

**Review by v-michaelmi — APPROVED** [review]

File: 6-AdvancedScenarios/3-call-api-acrs/API/models/authContext.js [hunk 2]
```diff
@@ -39,9 +39,9 @@ class AuthContext {
         db.get('acrs').push(newAcrs).write();
     }
 
-    static deleteAuthContext(authContextId) {
+    static deleteAuthContext(authContextObject) {
         db.get('acrs')
-            .remove({ authContextId: authContextId })
+            .remove({ authContextId: authContextObject.authContextId,  operation: authContextObject.operation})
```
Comment (v-michaelmi) [inline 6-AdvancedScenarios/3-call-api-acrs/API/models/authContext.js:44]: Nit:

You should be able to use object destructuring here.

```javascript
        db.get('acrs')
            .remove({ authContextId })
            .remove({ authContextObject.authContextId, authContextObject.operation})
```

File: 6-AdvancedScenarios/3-call-api-acrs/API/utils/routeGuard.js [hunk 3]
```diff
@@ -3,18 +3,23 @@ const AuthContext = require('../models/authContext');
...
     if (acrs.length === 0) {
         return next();
     } else {
-        const authContext = acrs.find(ac => ac.operation === req.method && ac.tenantId === req.authInfo.tid); 
+        const authContext = acrs.find((ac) => {
+            if(isPut){
+                return ac.operation === 'UPDATE' && ac.tenantId === req.authInfo.tid
+            }
+            return ac.operation === req.method && ac.tenantId === req.authInfo.tid
+        });
```
Comment (v-michaelmi) [inline 6-AdvancedScenarios/3-call-api-acrs/API/utils/routeGuard.js:17]: Not sure why this was deleted.

```javascript
const authContext = acrs
	.filter(ac => ac.tenantId === req.authInfo.tid)
	.find(ac => isPut ? ac.operation === 'UPDATE' : ac.operation === req.method);
```

Comment (v-michaelmi) [reply 6-AdvancedScenarios/3-call-api-acrs/API/utils/routeGuard.js:17]: After looking at the whole function we can tidy up a bit more.

```javascript
const authContextGuard = (req, res, next) => {
    const acrs = AuthContext.getAuthContexts();
    const isPut = req.method === "PUT";

    const authContext = acrs
	.filter(ac => ac.tenantId === req.authInfo.tid)
	.find(ac => isPut ? ac.operation === 'UPDATE' : ac.operation === req.method);

    if (!!authContext) {
        // if found, check the request for the required claims
        return checkForRequiredAuthContext(req, res, next, authContext.authContextId);
    }

    // if there is no auth context in the db, let the request through
    next();
}
```

**Pull request closed without merging**